packages/Webkul/Admin/tests/Feature/Catalog/ProductTest.php

```diff
@@ -40,64 +40,699 @@
+        'type'                => 'simple',
         'attribute_family_id' => 1,
         'sku'                 => $sku = fake()->uuid(),
     ])
         ->assertOk()
         ->assertJsonPath('data.redirect_url', route('admin.catalog.products.edit', $productId));
 
     $this->assertDatabaseHas('products', [
-        'id'   => $productId,
-        'type' => $type,
+        'type' => 'simple',
         'sku'  => $sku,
+        'id'   => $productId,
     ]);
 });
 
-it('should copy the existing product', function () {
+it('should return the edit page of simple product', function () {
+    // Arrange
+    $product = (new ProductFaker())->getSimpleProductFactory()->create();
+
+    // Act and Asssert
+    $this->loginAsAdmin();
+
+    $this->get(route('admin.catalog.products.edit', $product->id))
+        ->assertOk()
+        ->assertSeeText(trans('admin::app.catalog.products.edit.title'))
+        ->assertSeeText(trans('admin::app.account.edit.back-btn'))
+        ->assertSeeText($product->url_key)
+        ->assertSeeText($product->name)
+        ->assertSeeText($product->short_description)
+        ->assertSeeText($product->description);
+});
+
+it('should update the simple product', function () {
     // Arrange
     $product = (new ProductFaker())->getSimpleProductFactory()->create();
 
+    // Act and Asssert
+    $this->loginAsAdmin();
+
+    putJson(route('admin.catalog.products.update', $product->id), [
+        'sku'               => $product->sku,
+        'url_key'           => $product->url_key,
+        'short_description' => $shortDescription = fake()->sentence(),
+        'description'       => $description = fake()->paragraph(),
+        'name'              => $name = fake()->words(3, true),
+        'price'             => $price = fake()->randomFloat(2, 1, 1000),
+        'weight'            => $weight = fake()->numberBetween(0, 100),
+        'channel'           => $channel = core()->getCurrentChannelCode(),
+        'locale'            => $locale = app()->getLocale(),
+    ])
+        ->assertRedirect(route('admin.catalog.products.index'))
+        ->isRedirection();
+
+    $this->assertDatabaseHas('products', [
+        'sku'  => $product->sku,
+        'type' => $product->type,
+    ]);
+
+    $this->assertDatabaseHas('product_flat', [
+        'url_key'              => $product->url_key,
+        'type'                 => 'simple',
+        'name'                 => $name,
+        'short_description'    => $shortDescription,
+        'description'          => $description,
+        'price'                => $price,
+        'weight'               => $weight,
+        'locale'               => $locale,
+        'product_id'           => $product->id,
+        'channel'              => $channel,
+    ]);
+});
+
+it('should return the create page of configurable product', function () {
     // Act and Assert
     $this->loginAsAdmin();
 
```

Comment:
// Arrange
$attributes = AttributeFamily::find(1)->configurable_attributes

then check this attributes in assertions rathr then hard coding

Reply:
done